swiftpkg/internal/swiftpkg_build_files.bzl

```diff
@@ -1053,6 +1057,12 @@ def _new_for_license(pkg_info, license):
         decls = decls,
     )
 
+# MARK: - Build files encapsulation
+
+def _build_file_visibility(repository_ctx):
+    experimental_expose_build_files = getattr(repository_ctx.attr, "experimental_expose_build_files", False)
+    return ["//visibility:public"] if experimental_expose_build_files else ["//:__subpackages__"]
+
```

Comment:
Instead of checking the flag value in this module, let's add `expose_build_files` to `pkg_info` [here](https://github.com/cgrindel/rules_swift_package_manager/blob/7f388fe833a6cc67ce74db13c7c2021b6bedfcb4/swiftpkg/internal/pkginfos.bzl#L633-L645). We should read the value [here](https://github.com/cgrindel/rules_swift_package_manager/blob/7f388fe833a6cc67ce74db13c7c2021b6bedfcb4/swiftpkg/internal/pkginfos.bzl#L604-L611) like we do the `remote` and `commit`. 

Once we do that, we can stop passing `repository_ctx` into the bowels of the build file generation logic.